source/Octopus.Tentacle.Kubernetes.Tests.Integration/KubernetesAgent/KubernetesAgentMigrateFromPreinstallationTest.cs

```diff
@@ -0,0 +1,189 @@
+using Octopus.Tentacle.Startup;
+
+namespace Octopus.Tentacle.Kubernetes.Tests.Integration.KubernetesAgent;
+
+public class KubernetesAgentMigrateFromPreinstallationTest
+{
+    
+    readonly ISystemLog systemLog = new SystemLog();
+    
+    [Test]
+    public async Task MigrationFromPreinstallHookShouldCopyData()
+    {
+        //Arrange
+        var kubernetesConfigClient = new KubernetesFileWrappedProvider(KubernetesTestsGlobalContext.Instance.KubeConfigPath);
+        var commandNamespace = Guid.NewGuid().ToString("N");
+        Environment.SetEnvironmentVariable(KubernetesConfig.NamespaceVariableName, commandNamespace);
+        var commandToRun = new MigratePreInstalledKubernetesDeploymentTargetCommand(new Lazy<IKubernetesClientConfigProvider>(kubernetesConfigClient), systemLog, new LogFileOnlyLogger());
+        var client = new k8s.Kubernetes(kubernetesConfigClient.Get());
+        var validationKey = Guid.NewGuid().ToString("N");
+        var sourceConfigMapData = new Dictionary<string, string>
+        {
+            {"validationKey", validationKey},
+            {"Tentacle.Services.IsRegistered", "true"}
+        };
+        var sourceSecretData = new Dictionary<string, string>
+        {
+            {"machine-iv", "testData"},
+            {"machine-key", validationKey}
+        };
+        const string sourceConfigMapName = "tentacle-config-pre";
+        const string sourceSecretName = "tentacle-secret-pre";
+        const string destinationConfigMapName = "tentacle-config";
+        const string destinationSecretName = "tentacle-secret";
+
+        
+        string[] commandArguments = [
+            $"--source-config-map-name={sourceConfigMapName}", 
+            $"--source-secret-name={sourceSecretName}",
+            $"--destination-config-map-name={destinationConfigMapName}",
+            $"--destination-secret-name={destinationSecretName}",
+            $"--namespace={commandNamespace}"
+        ];
+
+        //Act
+        await CreateCommandNamespace(client, commandNamespace);
+        
+        //Sources
+        await CreateConfigmap(client, sourceConfigMapName, commandNamespace, sourceConfigMapData);
+        await CreateSecret(client, sourceSecretName, commandNamespace, sourceSecretData);
+
+        
+        // Targets
+        await CreateConfigmap(client, destinationConfigMapName, commandNamespace);
+        await CreateSecret(client, destinationSecretName, commandNamespace);
+        
+        commandToRun.Start(commandArguments, new NoninteractiveHost(), []);
+        
+        //Assert
+        var configMap = await client.CoreV1.ReadNamespacedConfigMapAsync(destinationConfigMapName, commandNamespace);
+        var secret = await client.CoreV1.ReadNamespacedSecretAsync(destinationSecretName, commandNamespace);
+
+        configMap.Data.TryGetValue("validationKey", out var validationKeyFromKubernetes);
+        validationKeyFromKubernetes.Should().NotBeNull().And.Be(validationKey);
```

Comment:
I'd have though you could skip the `NoBeNull` since you're asserting that it is a value